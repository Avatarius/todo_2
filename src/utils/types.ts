interface ITodo {
  id: string;
  text: string;
  isCompleted: boolean;
}

enum FilterType {
  All,
  Active,
  Completed
}


export type {ITodo};
export {FilterType};
