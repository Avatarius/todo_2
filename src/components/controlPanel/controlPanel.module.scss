@import "../../styles/mixins/button.scss";
.container {
  display: grid;
  grid-template-columns: 1fr 2fr 1fr;
  justify-content: center;
  align-items: center;
  padding-inline: 7px;
  font-size: 14px;
  padding-block: 10px;
  background-color: #e7e7e7;
}

.switch {
  display: flex;
  justify-content: center;
  align-items: center;
  gap: 5px;

  &:hover {

  }
  &__button {
    @include button;
    padding: 5px;
    &:hover {
      border: 1px solid #7777773a;
    }
    &_active {
      border: 1px solid var(--border-color, #77777780);
    }
  }
}

.clear {
  @include button;
}

.info {
  display: block;
}
