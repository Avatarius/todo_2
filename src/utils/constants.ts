import { ITodo } from "./types";

const initialData: ITodo[] = [
  {
    id: crypto.randomUUID(),
    text: "Тестовое задание",
    isCompleted: false,
  },
  {
    id: crypto.randomUUID(),
    text: "Прекрасный код",
    isCompleted: true,
  },
  {
    id: crypto.randomUUID(),
    text: "Покрытие тестами",
    isCompleted: false,
  },
];

export{initialData};
