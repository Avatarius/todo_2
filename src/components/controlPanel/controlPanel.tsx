import clsx from "clsx";
import styles from "./controlPanel.module.scss";
import { FilterType, ITodo } from "../../utils/types";

interface IControlPanelProps {
  length: number;
  filter: FilterType;
  callbacks: {
    all: () => void;
    active: () => void;
    completed: () => void;
    clear: () => void;
  };
}

function ControlPanel(props: IControlPanelProps) {
  const { length, filter, callbacks: {all, active, completed, clear} } = props;
  return (
    <div className={styles.container}>
      <span className={styles.info}>{length} items left</span>
      <div className={styles.switch}>
        <button
          className={clsx(styles.switch__button, filter === FilterType.All && styles.switch__button_active)}
          onClick={all}
        >
          All
        </button>
        <button className={clsx(styles.switch__button, filter === FilterType.Active && styles.switch__button_active)} onClick={active}>Active</button>
        <button className={clsx(styles.switch__button, filter === FilterType.Completed && styles.switch__button_active)} onClick={completed}>Completed</button>
      </div>
      <button className={styles.clear} onClick={clear}>Clear completed</button>
    </div>
  );
}

export { ControlPanel };
