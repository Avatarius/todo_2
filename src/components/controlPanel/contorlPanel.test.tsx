import renderer from "react-test-renderer";
import { expect, test, describe } from "@jest/globals";
import { ControlPanel } from "./controlPanel";
import { FilterType } from "../../utils/types";

describe("Компонент <ControlPanel/>", () => {
  test("Рендер компонента", () => {
    const component = renderer.create(
      <ControlPanel
        length={3}
        filter={FilterType.Completed}
        callbacks={{
          all: () => {},
          active: () => {},
          completed: () => {},
          clear: () => {},
        }}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
