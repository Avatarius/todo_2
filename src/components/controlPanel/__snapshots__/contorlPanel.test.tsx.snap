// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент <ControlPanel/> Рендер компонента 1`] = `
<div>
  <span>
    3
     items left
  </span>
  <div>
    <button
      className=""
      onClick={[Function]}
    >
      All
    </button>
    <button
      className=""
      onClick={[Function]}
    >
      Active
    </button>
    <button
      className=""
      onClick={[Function]}
    >
      Completed
    </button>
  </div>
  <button
    onClick={[Function]}
  >
    Clear completed
  </button>
</div>
`;
